tracpro/templates/lang_switcher.html

```diff
@@ -1,18 +1,34 @@
-<ul class="lang-switcher">
-  {% for code, language in available_languages %}
-    <li>
-      {% if code == request.LANGUAGE_CODE %}
-        {{ language }}
-      {% else %}
+{% load i18n %}
+<ul class="lang-switcher list-inline">
+  {% for code, language in all_languages %}
+    {% if code == request.LANGUAGE_CODE %}
+      <li>{{ language }}</li>
+    {% else %}
+      <li{% if code not in show_languages %} class="hidden"{% endif %}>
         <form action="{% url "set_language" %}" method="POST">
           {% csrf_token %}
           <input type="hidden" name="next" value="{{ request.path }}" />
           <input type="hidden" name="language" value="{{ code }}" />
-          <a class="lang-submit" href="#" onclick="$(this).closest('form').submit();">
+          <button class="btn btn-link" type="submit">
```

Comment:
Good improvement here, switching from anchor to submit button. Simplified the code nicely.